doc/source/reference/sparse.migration_to_sparray.rst

```diff
@@ -245,17 +314,160 @@ Removed methods and attributes
+
+        class _strict_bsr_matrix(_strict_mul_mixin, scipy.sparse.bsr_matrix):
+            pass
+
+        class _strict_csr_matrix(_strict_mul_mixin, scipy.sparse.csr_matrix):
+            pass
+
+        class _strict_csc_matrix(_strict_mul_mixin, scipy.sparse.csc_matrix):
+            pass
+
+        class _strict_dok_matrix(_strict_mul_mixin, scipy.sparse.dok_matrix):
+            pass
+
+        class _strict_lil_matrix(_strict_mul_mixin, scipy.sparse.lil_matrix):
+            pass
+
+        class _strict_dia_matrix(_strict_mul_mixin, scipy.sparse.dia_matrix):
+            pass
+
+        scipy.sparse.coo_matrix = scipy.sparse._coo.coo_matrix = _strict_coo_matrix
+        scipy.sparse.bsr_matrix = scipy.sparse._bsr.bsr_matrix = _strict_bsr_matrix
+        scipy.sparse.csr_matrix = scipy.sparse._csr.csr_matrix = _strict_csr_matrix
+        scipy.sparse.csc_matrix = scipy.sparse._csc.csc_matrix = _strict_csc_matrix
+        scipy.sparse.dok_matrix = scipy.sparse._dok.dok_matrix = _strict_dok_matrix
+        scipy.sparse.lil_matrix = scipy.sparse._lil.lil_matrix = _strict_lil_matrix
+        scipy.sparse.dia_matrix = scipy.sparse._dia.dia_matrix = _strict_dia_matrix
+
+        scipy.sparse._compressed.csr_matrix = _strict_csr_matrix
+
+        scipy.sparse._construct.bsr_matrix = _strict_bsr_matrix
+        scipy.sparse._construct.coo_matrix = _strict_coo_matrix
+        scipy.sparse._construct.csc_matrix = _strict_csc_matrix
+        scipy.sparse._construct.csr_matrix = _strict_csr_matrix
+        scipy.sparse._construct.dia_matrix = _strict_dia_matrix
+
+        scipy.sparse._extract.coo_matrix = _strict_coo_matrix
+
+        scipy.sparse._matrix.bsr_matrix = _strict_bsr_matrix
+        scipy.sparse._matrix.coo_matrix = _strict_coo_matrix
+        scipy.sparse._matrix.csc_matrix = _strict_csc_matrix
+        scipy.sparse._matrix.csr_matrix = _strict_csr_matrix
+        scipy.sparse._matrix.dia_matrix = _strict_dia_matrix
+        scipy.sparse._matrix.dok_matrix = _strict_dok_matrix
+        scipy.sparse._matrix.lil_matrix = _strict_lil_matrix
+
+
+.. _sparse-migration-index-array-dtypes:
+
+Index Array DTypes
+------------------
+
+If you provide compressed indices to a constructor,
+e.g. ``csr_array((data, indices, indptr))`` sparse arrays set the index
+dtype by only checking the index arrays dtype, while sparse matrices check the
+index values too and may downcast to int32
+(see `gh-18509 <https://github.com/scipy/scipy/pull/18509>`__ for more details).
+This means you may get int64 indexing when you used to get int32.
+You can control this by setting the ``dtype`` before instantiating, or by
+recasting after construction.
+
+Two sparse utility functions can help with handling the index dtype.
+Use ``get_index_dtype(arrays, maxval, check_contents)`` while creating indices
+to find an appropriate dtype (int32 or int64) to use for your compressed indices.
+
+Use ``scipy.sparse.safely_cast_index_arrays`` for recasting after construction,
```

Comment:
As of gh-21858, this isn't yet exposed publicly, right?

Reply:
Good catch!
I edited this line to remove the path to the function (just like the description of `get_index_dtype` above).
And then I edited the code section below to include the path for both of these helper functions.

Thanks!!